File: app/src/main/java/com/example/myanimedata/api/ImageResponse.java
package com.example.myanimedata.api;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

public class ImageResponse {
    @SerializedName("data")
    private final ArrayList<ImageResult> imageResultsList;

    public ArrayList<ImageResult> getImageResultsList() {
        return imageResultsList;
    }

    public ImageResponse(ArrayList<ImageResult> imageResultsList) {
        this.imageResultsList = imageResultsList;
    }
}

File: app/src/main/java/com/example/myanimedata/api/ImageResponse.kt
package com.example.myanimedata.api

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

@Parcelize
data class ImageResponse(
    @field:SerializedName("data") val imageResultsList: ArrayList<ImageResult>? = null
) : Parcelable